pydicom/dataset.py

```diff
@@ -475,13 +475,12 @@ def __contains__(self, name):
             ``False`` otherwise.
         """
         try:
-            tag = Tag(name)
-        except (ValueError, OverflowError):
-            return False
-        # Test against None as (0000,0000) is a possible tag
-        if tag is not None:
-            return tag in self._dict
-        return name in self._dict  # will no doubt raise an exception
+            return Tag(name) in self._dict
+        except Exception as exc:
```

Comment:
Maybe better catch concrete excptions here; and maybe vary the error messages accordingly - for example for overflow error the current message may not fit.